app/models/katello/host/content_facet.rb

```diff
@@ -164,10 +164,12 @@ def update_repositories_by_paths(paths)
         prefixes = %w(/pulp/deb/ /pulp/repos/ /pulp/content/)
         relative_paths = []
 
-        # paths == ["/pulp/content/Default_Organization/Library/custom/Test_product/test2"]
+        # paths == ["/pulp/content/Default_Organization/Library/custom/Test_product/test2",
+        #           "/pulp/content/Org/Library/custom/Test_product/test2/%3Fcomp%3Dmain%26rel%3Dstable"]
         paths.each do |path|
           if (prefix = prefixes.find { |pre| path.start_with?(pre) })
-            relative_paths << path.gsub(prefix, '')
+            # strip prefix and structured_apt postfix before adding to relative_paths
+            relative_paths << path.sub(prefix, '').sub(/\/?%3F.*/, '')
```

Comment:
This fixes an issue we identified during review/testing.

The subman will report `baseurl` of enabled_repositories from `/etc/apt/sources.list.d/rhsm.sources`.
With structured APT, these `baseurl`s includes query-parameters to transfer release and components.
Katello would not be able to find the corresponding repositories without first stripping the query-parameters.

I also changed the `gsub()` that strips the prefix to `sub()`. It is not probable that the prefix is also part of a repository's `relative_path`, but better be safe than sorry :wink: